File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:

        largestVolume = 0
        lp = 0
        rp = len(heights) - 1
        
        toggle = True
        while lp != rp:
        
            length = rp - lp
            minSide = min(heights[lp],heights[rp])

            currVolume = length * minSide
            largestVolume = max(largestVolume, currVolume)

            if toggle:
                lp += 1     
            else:
                rp -= 1    

            
            toggle = not toggle  

        
        return largestVolume

File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights) {

        int p1 = 0;
        int p2 = heights.length - 1;
        int result = 0;

        while (p1 < p2){
            int width = p2 - p1;
            int height = Math.min(heights[p2], heights[p1]);
            int area = width * height;
            result = Math.max(result, area);

            if (heights[p1] > heights[p2]){
                p2--;
            } else if (heights[p2]> heights[p1]){
                p1++;
            } else{
                if (heights[p2-1] > heights[p1+1]){
                    p2--;
                } else{
                    p1++;
                }
            }
        }

        return result;
        
    }
}
